### PR TITLE
added demo page

### PR DESCRIPTION
**Pull request by Nikcodes10**
## Issue that this pull request solves

 Closes: #195 

### Brief description of what is fixed or changed

Added Demo page
Home page now contains 2 buttons: Demo and Pro
Added Go back button to Navbar
Displays Logout button in Navbar only when user is logged-in
Demo page shows alert and hence become not-usable after 60 seconds.

## Types of changes

_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (Documentation content changed)
- [ ] Other (please describe): 

## Checklist

_Put an `x` in the boxes that apply_

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] My changes does not break the current system and it passes all the current test cases.

## Screenshots

Please attach the screenshots of the changes made in case of change in user interface

Screenshot (344)
Screenshot (345)
Screenshot (346)

Any other information that is important to this pull request

I am a DSOC and SWOC participant

### CODE REVIEW

File: src/app/components/home/home.component.html [hunk 1]
```diff
@@ -40,7 +40,10 @@
 
     <div class="row justify-content-center">
       <div class="demo-button mt-3" data-aos="zoom-in-up" data-aos-duration="1000">
-        <button class="btn-grad" routerLink="/login" >Try it out!</button>
+        <div class="row">
+          <button class="btn-grad" routerLink="/demo" >Demo</button>
+          <button class="btn-grad" routerLink="/login" >Pro</button>
```
Comment (goliakshay357) [inline src/app/components/home/home.component.html:45]: Hey, 
There is nothing like pro in canvasboard. "TRY IT OUT" button will directly go to demo page (i.e demo component)

**Comment by goliakshay357** [comment]
Your build is failing. Please checkout in Details. 
Thanks

**Comment by kingavatar** [comment]
Can you also please provide gif or mp4 preview of your UI changes. Thank you.

**Comment by Nikcodes10** [comment]
Ok

**Comment by Nikcodes10** [comment]
Removed demo button
Try it out button goes to demo page
Added Sign-in button to Navbar

Screenshot (349)
Screenshot (348)

Why is the build failing?

Only reason I can see:

Warning: that 2mb budget exceeded

Screenshot (347)

Do run locally and let me know if anything is wrong.

**Comment by goliakshay357** [comment]
Hello @Nikcodes10,
The build is failing at this stage (see attached scr shot)
image

Just confirming, did you take the latest pull before making a PR?
Thanks

**Comment by Kajol-Kumari** [comment]
Hey @Nikcodes10 do checkout [here](https://github.com/Canvasbird/canvasboard/pull/328/checks?check_run_id=1702724880#step:6:20) Seems like you forgot to import it in the test file of the `demoComponent` and that's making the test to fail. Check this once and it should be good to go!

**Comment by Nikcodes10** [comment]
@Kajol-Kumari I tried that. But it throws an error regarding the import of the Modules such as Location, HttpClient and Router. 

The problem is simple: Importing modules like AuthService, Router and Location into constructor of Demo component throws an error during tests. 

I don't know how to fix it.

I have fixed it for now by copying the navbar code and making it non-functional. In the ngOnInit function, I have commented what to do.

**Comment by Kajol-Kumari** [comment]
Hey @Nikcodes10 I won't recommend copy-pasting the whole code instead of selector. Just make sure to follow [this](https://stackoverflow.com/questions/43937387/if-selector-is-an-angular-component-then-verify-that-it-is-part-of-this-mod) and your build should pass. Then this PR will be good to go, we can ignore the deepsource error if it doesn't get fixed after the build fix

**Comment by Nikcodes10** [comment]
Fixed everything. 
It was a simple problem of not importing the HttpClientTestingModule and the RouterTestingModule into the spec.ts files of demo and navbar components.

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]
Hey @Nikcodes10 please do tye following-
1. Take the latest pull and resolve the conflicts. 
2. Remove unnecessary file changes like that if `debug.log` and `package-lock.json`

**Comment by Nikcodes10** [comment]
@Kajol-Kumari 
DWOC

**Pull request closed without merging**